From b6d6f2ff400c7824574ffead18585aadb60a02ef Mon Sep 17 00:00:00 2001
From: Shayne Chen <shayne.chen@mediatek.com>
Date: Thu, 19 Oct 2023 10:51:55 +0800
Subject: [PATCH 63/69] mtk: wpa_s: force MLD STA to use SAE H2E during
 authentication

Otherwise the MLD STA setup will fail with hostapd MLD AP.

Signed-off-by: Shayne Chen <shayne.chen@mediatek.com>
---
 wpa_supplicant/sme.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wpa_supplicant/sme.c b/wpa_supplicant/sme.c
index f08184f98..e1183722f 100644
--- a/wpa_supplicant/sme.c
+++ b/wpa_supplicant/sme.c
@@ -199,7 +199,7 @@ static struct wpabuf * sme_auth_build_sae_commit(struct wpa_supplicant *wpa_s,
 	if (wpa_key_mgmt_sae_ext_key(key_mgmt) &&
 	    wpa_s->conf->sae_pwe != SAE_PWE_FORCE_HUNT_AND_PECK)
 		use_pt = 1;
-	if (bss && is_6ghz_freq(bss->freq) &&
+	if (bss && (is_6ghz_freq(bss->freq) || !is_zero_ether_addr(bss->mld_addr)) &&
 	    wpa_s->conf->sae_pwe != SAE_PWE_FORCE_HUNT_AND_PECK)
 		use_pt = 1;
 #ifdef CONFIG_SAE_PK
-- 
2.39.2

